<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FacedistortedClass</class>
 <widget class="QMainWindow" name="FacedistortedClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1164</width>
    <height>779</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Facedistorted</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_9">
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>373</width>
           <height>339</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QScrollArea" name="scrollArea_2">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents_2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>374</width>
           <height>339</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QScrollArea" name="scrollArea_3">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>373</width>
           <height>339</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_11">
      <item>
       <widget class="QGroupBox" name="groupBox_8">
        <property name="title">
         <string>特征点显示</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QScrollArea" name="scrollArea_4">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents_4">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>335</width>
              <height>262</height>
             </rect>
            </property>
           </widget>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QPushButton" name="pushButton_9">
             <property name="text">
              <string>原图</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_10">
             <property name="text">
              <string>目标图</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>变形函数</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QRadioButton" name="radioButton">
             <property name="text">
              <string>TPS</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_2">
             <property name="text">
              <string>B样条（小）</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_12">
             <property name="text">
              <string>B样条（大）</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_13">
             <property name="text">
              <string>B样条+TPS</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>变形方向</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QRadioButton" name="radioButton_3">
             <property name="text">
              <string>逆向</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_4">
             <property name="text">
              <string>正向最近邻</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QGroupBox" name="groupBox_7">
            <property name="title">
             <string>图像调整</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QRadioButton" name="radioButton_9">
               <property name="text">
                <string>移动</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="radioButton_8">
               <property name="text">
                <string>放缩</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_9">
            <property name="title">
             <string>B样条网格大小</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QLineEdit" name="lineEdit">
               <property name="placeholderText">
                <string>30</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>逆向插值方式</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QRadioButton" name="radioButton_5">
             <property name="text">
              <string>最近邻</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_7">
             <property name="text">
              <string>双三次</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_6">
             <property name="text">
              <string>双线性</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_11">
             <property name="text">
              <string>Lanczos</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_10">
             <property name="text">
              <string>反距离</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>原图</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QPushButton" name="pushButton">
             <property name="text">
              <string>导入原图</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_5">
             <property name="text">
              <string>读取特征点</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_6">
             <property name="text">
              <string>自动检测特征点</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>目标图</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QPushButton" name="pushButton_2">
             <property name="text">
              <string>导入目标图</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_7">
             <property name="text">
              <string>读取特征点</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_8">
             <property name="text">
              <string>自动检测特征点</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>生成图</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QPushButton" name="pushButton_3">
             <property name="text">
              <string>开始</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_4">
             <property name="text">
              <string>保存</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1164</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="facedistorted.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>OpenImage1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>775</x>
     <y>506</y>
    </hint>
    <hint type="destinationlabel">
     <x>223</x>
     <y>276</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>OpenImage2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>775</x>
     <y>601</y>
    </hint>
    <hint type="destinationlabel">
     <x>443</x>
     <y>275</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>ImageCreat()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>775</x>
     <y>695</y>
    </hint>
    <hint type="destinationlabel">
     <x>557</x>
     <y>531</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>ImageSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>936</x>
     <y>695</y>
    </hint>
    <hint type="destinationlabel">
     <x>580</x>
     <y>581</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>ReadNum1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>850</x>
     <y>506</y>
    </hint>
    <hint type="destinationlabel">
     <x>810</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>ExtractNum1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>952</x>
     <y>506</y>
    </hint>
    <hint type="destinationlabel">
     <x>946</x>
     <y>390</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>ReadNum2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>828</x>
     <y>601</y>
    </hint>
    <hint type="destinationlabel">
     <x>935</x>
     <y>574</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>ExtractNum2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>925</x>
     <y>601</y>
    </hint>
    <hint type="destinationlabel">
     <x>938</x>
     <y>614</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_9</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>ShowNum1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>695</y>
    </hint>
    <hint type="destinationlabel">
     <x>94</x>
     <y>717</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_10</sender>
   <signal>clicked()</signal>
   <receiver>FacedistortedClass</receiver>
   <slot>ShowNum2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>223</x>
     <y>699</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>725</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OpenImage1()</slot>
  <slot>OpenImage2()</slot>
  <slot>slot1()</slot>
  <slot>open()</slot>
  <slot>ImageCreat()</slot>
  <slot>ImageSave()</slot>
  <slot>ReadNum1()</slot>
  <slot>ExtractNum1()</slot>
  <slot>ReadNum2()</slot>
  <slot>ExtractNum2()</slot>
  <slot>ZoomNum()</slot>
  <slot>ShowNum1()</slot>
  <slot>ShowNum2()</slot>
 </slots>
</ui>
